File: platforms/ios/vshop/farwolf.weex/farwolf.weex/module/WXNetModule.m
//
//  WXNetModule.m
//  Pods
//
//  Created by 郑江荣 on 2017/5/19.
//
//

#import "WXNetModule.h"

@implementation WXNetModule
@synthesize weexInstance;
WX_EXPORT_METHOD(@selector(post:param:header:start:success:exception:compelete:))
WX_EXPORT_METHOD(@selector(get:param:header:start:success:exception:compelete:))

-(void)fetch:(BOOL)usepost url:(NSString*)url param:(NSDictionary*)param header:(NSDictionary*)header start:(WXModuleKeepAliveCallback)start exception:(WXModuleKeepAliveCallback)exception success:(WXModuleKeepAliveCallback)success compelete:(WXModuleKeepAliveCallback)compelete
{
    JsonReader *j=[JsonReader new];
    j.url=url;
    j.header=header;
    j.param=param;
    [j excuteNoLimit:^{
        start(@{},false);
    } success:^(Json *j) {
        success(@{@"res":j.data,@"sessionid":j.sessionId},false);
    } exception:^{
        exception(@{},false);
    } compelete:^{
         compelete(@{},false);
    } usePost:usepost];
}




-(void)post:(NSString*)url param:(NSDictionary*)param header:(NSDictionary*)header start:(WXModuleKeepAliveCallback)start  success:(WXModuleKeepAliveCallback)success exception:(WXModuleKeepAliveCallback)exception compelete:(WXModuleKeepAliveCallback)compelete
{
    [self fetch:true url:url param:param header:header start:start exception:exception success:success compelete:compelete];
}


-(void)get:(NSString*)url param:(NSDictionary*)param header:(NSDictionary*)header start:(WXModuleKeepAliveCallback)start  success:(WXModuleKeepAliveCallback)success exception:(WXModuleKeepAliveCallback)exception compelete:(WXModuleKeepAliveCallback)compelete
{
    [self fetch:false url:url param:param header:header start:start exception:exception success:success compelete:compelete];
}

@end

File: platforms/android/farwolf.weex/src/main/java/com/farwolf/weex/module/WXNetModule.java
package com.farwolf.weex.module;

import com.farwolf.interfac.IHttp;
import com.farwolf.volley.VolleyReq;
import com.taobao.weex.annotation.JSMethod;
import com.taobao.weex.bridge.JSCallback;
import com.taobao.weex.common.WXModule;

import java.util.HashMap;

/**
 * Created by zhengjiangrong on 2017/5/22.
 */

public class WXNetModule extends WXModule {



    public void fetch(boolean usepost, final  String url, HashMap param, HashMap header, final JSCallback start, final JSCallback success, final JSCallback compelete, final JSCallback exception)
    {

       final VolleyReq v=new VolleyReq(mWXSDKInstance.getContext()){

            @Override
            public String getUrl() {
                return url;
            }
        };
        v.addParams(param);
        v.addHeaders(header);


        v.excute(new IHttp() {
            @Override
            public void OnPostStart(Object o) {
                start.invoke(o);
            }

            @Override
            public void OnPostCompelete(Object o) {
                HashMap m=new HashMap();
                m.put("res",o);
                m.put("sessionid",v.getCookie());
                success.invoke(m);
                compelete.invoke(o);
            }

            @Override
            public void OnException(Object o) {
                 exception.invoke(o);
                compelete.invoke(o);
            }
        },usepost);




    }

    @JSMethod
    public void post(String url , HashMap param, HashMap header, final JSCallback start, final JSCallback success, final JSCallback compelete, final JSCallback exception)
    {
        this.fetch(true,url,param,header,start,success,compelete,exception);
    }

    @JSMethod
    public void get(String url , HashMap param, HashMap header, final JSCallback start, final JSCallback success, final JSCallback compelete, final JSCallback exception)
    {
        this.fetch(false,url,param,header,start,success,compelete,exception);
    }

}
